packages/backend/src/managers/environmentManager.ts

```diff
@@ -131,4 +136,42 @@ export class EnvironmentManager {
         console.error(`Something went wrong while emitting MSG_ENVIRONMENTS_STATE_UPDATE: ${String(err)}`);
       });
   }
+
+  async deleteEnvironment(recipeId: string) {
+    try {
+      this.setEnvironmentStatus(recipeId, 'stopping');
+      const envPod = await this.getEnvironmentPod(recipeId);
+      await containerEngine.stopPod(envPod.engineId, envPod.Id);
+      this.setEnvironmentStatus(recipeId, 'removing');
+      await containerEngine.removePod(envPod.engineId, envPod.Id);
```

Comment:
The object stored in #environments is not removed ?

Reply:
No, it is removed when the pod is detected to be removed (on forgetPod). I prefer to centralize the state management by detecting the events on pod, and not touch it internally